metricbeat/module/redis/info/data.go

```diff
@@ -7,11 +7,96 @@ import (
+		},
+		"cluster": common.MapStr{
+			"cluster_enabled": info["cluster_enabled"],
+		},
+		"cpu": common.MapStr{
+			"used_cpu_sys":           info["used_cpu_sys"],
+			"used_cpu_user":          info["used_cpu_user"],
+			"used_cpu_sys_children":  info["used_cpu_sys_children"],
+			"used_cpu_user_children": info["used_cpu_user_children"],
+		},
+		"memory": common.MapStr{
+			"used_memory":      info["used_memory"],
+			"used_memory_rss":  info["used_memory_rss"],
+			"used_memory_peak": info["used_memory_peak"],
+			"used_memory_lua":  info["used_memory_lua"],
+			"mem_allocator":    info["mem_allocator"], // Could be moved server as it rarely changes
+		},
+		"presistence": common.MapStr{
+			"loading":                      info["loading"],
+			"rdb_changes_since_last_save":  info["rdb_changes_since_last_save"],
+			"rdb_bgsave_in_progress":       info["rdb_bgsave_in_progress"],
+			"rdb_last_save_time":           info["rdb_last_save_time"],
+			"rdb_last_bgsave_status":       info["rdb_last_bgsave_status"],
+			"rdb_last_bgsave_time_sec":     info["rdb_last_bgsave_time_sec"],
+			"rdb_current_bgsave_time_sec":  info["rdb_current_bgsave_time_sec"],
+			"aof_enabled":                  info["aof_enabled"],
+			"aof_rewrite_in_progress":      info["aof_rewrite_in_progress"],
+			"aof_rewrite_scheduled":        info["aof_rewrite_scheduled"],
+			"aof_last_rewrite_time_sec":    info["aof_last_rewrite_time_sec"],
+			"aof_current_rewrite_time_sec": info["aof_current_rewrite_time_sec"],
+			"aof_last_bgrewrite_status":    info["aof_last_bgrewrite_status"],
+			"aof_last_write_status":        info["aof_last_write_status"],
+		},
+		"replication": common.MapStr{
+			"role":                           info["role"],
+			"connected_slaves":               info["connected_slaves"],
+			"master_repl_offset":             info["master_repl_offset"],
+			"repl_backlog_active":            info["repl_backlog_active"],
+			"repl_backlog_size":              info["repl_backlog_size"],
+			"repl_backlog_first_byte_offset": info["repl_backlog_first_byte_offset"],
+			"repl_backlog_histlen":           info["repl_backlog_histlen"],
+		},
+		"server": common.MapStr{
+			"redis_version":     info["redis_version"],
+			"redis_git_sha1":    info["redis_git_sha1"],
+			"redis_git_dirty":   info["redis_git_dirty"],
+			"redis_build_id":    info["redis_build_id"],
+			"redis_mode":        info["redis_mode"],
+			"os":                info["os"],
+			"arch_bits":         info["arch_bits"],
+			"multiplexing_api":  info["multiplexing_api"],
+			"gcc_version":       info["gcc_version"],
+			"process_id":        info["process_id"],
+			"run_id":            info["run_id"],
+			"tcp_port":          info["tcp_port"],
+			"uptime_in_seconds": info["uptime_in_seconds"],
+			"uptime_in_days":    info["uptime_in_days"],
+			"hz":                info["hz"],
+			"lru_clock":         info["lru_clock"],
+			"config_file":       info["config_file"],
```

Comment:
@urso you forget the few fields which I didn't include. Of course I could remove these fields manually instead. One other potential issue with automating this is the mapping. If we set it manually, we know exactly which fields are sent and can set the mapping. If we don't and new fields are added on the server side (for example newer version of redis), new fields are sent without mapping.